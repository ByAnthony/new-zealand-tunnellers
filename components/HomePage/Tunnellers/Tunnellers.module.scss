@use "app/variables.module.scss";

.intro {
  @include variables.component-margin;

  margin-top: variables.$base-margin;

  @include variables.desktop {
    margin-left: 0;
    margin-right: 0;
  }

  svg {
    background-image: url("../../../public/images/homepage/ART_3670.jpg");
    background-position: 20% 75%;
    background-size: cover;
    background-attachment: fixed;

    @include variables.tablet {
      background-position: top;
      background-position: center;
      background-size: contain;
    }

    @include variables.desktop {
      height: 540px;
      background-position: top;
      background-position: center;
      background-size: contain;
    }

    text {
      font-size: 1.2em;
      font-weight: 900;
      color: transparent;
      text-transform: uppercase;

      @include variables.tablet {
        font-size: 0.8em;
      }

      @include variables.desktop {
        font-size: 0.8em;
      }
    }
  }
}

.roll-wrapper {
  display: flex;
  justify-content: flex-end;
}

.discover {
  display: block;
  margin-bottom: variables.$xxxs-margin;
  font-size: 0.9rem;
  font-weight: 200;
  text-align: start;

  @include variables.tablet {
    font-size: 1rem;
  }

  @include variables.desktop {
    font-size: 1.1rem;
  }
}

.roll-container {
  width: fit-content;
  display: flex;
  flex-direction: column;
}

.roll {
  @include variables.component-margin;

  padding-bottom: variables.$xxxs-margin;
  font-size: 1.2rem;
  font-weight: 500;
  color: variables.$text-color;
  border-bottom: 1px solid variables.$secondary-color;
  transition:
    color 0.9s ease-in-out,
    border-bottom 0.9s ease-in-out;

  @include variables.tablet {
    padding-bottom: variables.$xxxs-margin;
    font-size: 1.3rem;
  }

  @include variables.desktop {
    padding-bottom: variables.$xxs-margin;
    font-size: 1.5rem;
  }

  &:hover {
    border-bottom: 1px solid variables.$secondary-color;
    color: variables.$secondary-color;
  }

  @include variables.desktop {
    margin-left: 0;
    margin-right: 0;
  }
}
