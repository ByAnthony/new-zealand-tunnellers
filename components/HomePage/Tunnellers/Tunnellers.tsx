"use client";

import Link from "next/link";

import { useWindowDimensions } from "@/utils/helpers/useWindowDimensions";

import STYLES from "./Tunnellers.module.scss";

export function Tunnellers() {
  const { width } = useWindowDimensions();

  const svgElement = (height: number, width: number) => {
    return (
      <svg viewBox={`0 0 100 ${height}`}>
        <defs>
          <mask id="mask" x="0" y="0" width="100" height={height}>
            <rect x="0" y="0" width="100" height={height} fill="white" />
            <text x="0" y="10" textAnchor="start">
              {width <= 512 ? (
                <>
                  <tspan x="0" dy="0.8em">
                    The Kiwis
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    who
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    fought
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    beneath
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    the no
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    man’s
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    land
                  </tspan>
                </>
              ) : (
                <>
                  <tspan x="0" dy="0.8em">
                    The Kiwis
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    who fought
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    beneath the
                  </tspan>
                  <tspan x="0" dy="0.8em">
                    no man’s land
                  </tspan>
                </>
              )}
            </text>
          </mask>
        </defs>
        <rect
          x="0"
          y="0"
          width="100"
          height={height}
          fill="rgb(24, 26, 27)"
          mask="url(#mask)"
        />
      </svg>
    );
  };

  return (
    <>
      <div className={STYLES.intro}>
        <div className={STYLES["intro-text"]}>
          {width && svgElement(width <= 512 ? 125 : 60, width)}
        </div>
      </div>
      <div className={STYLES["roll-wrapper"]}>
        <div className={STYLES["roll-container"]}>
          <Link href="/tunnellers/" className={STYLES.roll}>
            <span className={STYLES.discover}>Discover</span>
            The New Zealand Tunnellers
          </Link>
        </div>
      </div>
    </>
  );
}
