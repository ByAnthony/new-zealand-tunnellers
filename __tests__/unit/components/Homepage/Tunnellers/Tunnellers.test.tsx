import { render, screen } from "@testing-library/react";

import { Tunnellers } from "@/components/HomePage/Tunnellers/Tunnellers";

describe("Tunnellers", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<Tunnellers />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the main title", () => {
    render(<Tunnellers />);

    const title = screen.getByRole("heading", {
      name: "The Kiwis who fought beneath the no manâ€™s land",
    });
    expect(title).toBeInTheDocument();
  });

  test("renders link to all Tunnellers page", () => {
    render(<Tunnellers />);

    const tunnellersLink = screen.getByRole("link", {
      name: "Discover The New Zealand Tunnellers",
    });
    expect(tunnellersLink).toHaveAttribute("href", "/tunnellers");
  });
});
