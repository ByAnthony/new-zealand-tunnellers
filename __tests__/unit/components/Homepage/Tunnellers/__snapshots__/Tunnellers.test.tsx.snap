// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tunnellers matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="intro"
  >
    <h1>
      The Kiwis who fought beneath the no manâ€™s land
    </h1>
  </div>
  <div
    class="roll-wrapper"
  >
    <div
      class="roll-container"
    >
      <a
        class="roll"
        href="/tunnellers"
      >
        <p
          class="discover"
        >
          Discover
        </p>
        The New Zealand Tunnellers
      </a>
    </div>
  </div>
</DocumentFragment>
`;
